### PR TITLE
Access risks and needs

### PR DESCRIPTION
**Pull request by jezhiggins**
Adds details of the assess-risks-and-needs services. 
Expands out the OASys context a little.

### CODE REVIEW

File: src/main/kotlin/model/AssessRisksAndNeeds.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    override fun defineRelationships() {
+      listOf(riskAssessmentUi, assessmentService)
+        .forEach { it.uses(HMPPSAuth.system, "authenticates via") }
+
+      assessmentService.uses(Delius.communityApi, "Gets offender and past-offence details from")
+      assessmentService.uses(PrepareCaseForCourt.courtCaseService, "Gets offender and offence details from")
+      assessmentService.uses(OASys.assessmentsApi, "get offender past assessment details from")
+      assessmentService.uses(OASys.system, "pushes offender assessment details into")
```
Comment (sldblog) [inline src/main/kotlin/model/AssessRisksAndNeeds.kt:60]: Interesting, how do we push those details? I'm wondering if there's a component missing from `OASys`.

**Review by sldblog — APPROVED** [review]
👍 LGTM, no blockes

File: src/main/kotlin/model/AssessRisksAndNeeds.kt (hunk 1 above)
Comment (jezhiggins) [reply src/main/kotlin/model/AssessRisksAndNeeds.kt:60]: We're building it at the moment!

Comment (sldblog) [reply src/main/kotlin/model/AssessRisksAndNeeds.kt:60]: Does it push directly to the DB?

Comment (jezhiggins) [reply src/main/kotlin/model/AssessRisksAndNeeds.kt:60]: God no! Pushing stuff straight into the tables would be a disaster! There's some work being done in OASys to provide, essentially, a programmatic API in the form of a number of stored procedures. On our side, we're mapping the data we collect into the OASys data structures, then invoking those stored procedures. 

More generally, OASys is its database and the database is OASys. The application is written using Apex (which is the essentially Oracle Forms for the browser), so the application code, the reference data, the application data, everything, all lives in the database.

**Pull request merged**